### PR TITLE
Copy demo deployment to production

### PR DESCRIPTION
**Pull request by EarthlingDavey**
This doesn't use www.justice.gov.uk as ingress yet.

### CODE REVIEW

File: deploy/production/deployment.tpl.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+              cpu: 500m
+              memory: 200Mi
+            limits:
+              cpu: 500m
+              memory: 200Mi
+
+        - name: fpm
+          image: "${ECR_URL}:${IMAGE_TAG_FPM}"
+          ports:
+            - containerPort: 9000
```
Comment (wilson1000) [inline deploy/production/deployment.tpl.yml:51]: We don't need this anymore - sockets are king

File: deploy/production/ingress.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: justice-gov-uk-production-ingress
+  namespace: justice-gov-uk-production
+  annotations:
+    external-dns.alpha.kubernetes.io/set-identifier: justice-gov-uk-production-ingress-justice-gov-uk-production-green
+    external-dns.alpha.kubernetes.io/aws-weight: "100"
+    nginx.ingress.kubernetes.io/auth-type: basic
+    nginx.ingress.kubernetes.io/auth-secret: basic-auth-secret
```
Comment (wilson1000) [inline deploy/production/ingress.yml:10]: We can remove basic auth, only we should reach it over host file redirect

File: deploy/production/ingress.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    nginx.ingress.kubernetes.io/auth-realm: 'Production User | Authentication Required'
+    nginx.ingress.kubernetes.io/proxy-body-size: "200m"
```
Comment (wilson1000) [inline deploy/production/ingress.yml:12]: Do we know what this means??

File: deploy/production/ingress.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      location = /.well-known/security.txt {
+        auth_basic off;
+        return 301 https://raw.githubusercontent.com/ministryofjustice/security-guidance/main/contact/vulnerability-disclosure-security.txt;
+      }
+spec:
+  ingressClassName: default
+  tls:
+  - hosts:
+    - production-justice-gov-uk.apps.live.cloud-platform.service.justice.gov.uk
+  # - hosts:
```
Comment (wilson1000) [inline deploy/production/ingress.yml:30]: We can update the local host file to reach this, is that what we discussed?

**Review by wilson1000 — COMMENTED** [review]
Looks ok @EarthlingDavey - some comments I'd like to chat about first 😄

File: deploy/production/ingress.yml (hunk 3 above)
Comment (EarthlingDavey) [reply deploy/production/ingress.yml:12]: proxy-body-size is for allowing posing documents (zips) < 200 megs.

Ref: https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#proxy-body-size

Looks like m (or M) is is ok to to use for megabytes https://nginx.org/en/docs/syntax.html

File: deploy/production/ingress.yml (hunk 4 above)
Comment (EarthlingDavey) [reply deploy/production/ingress.yml:30]: It was just a thought...

having `production-justice-gov-uk.apps.live.cloud-platform.service.justice.gov.uk` is one way to get the content up for QA.

Then prior to switchover, we change WP_HOME etc. and do a find/replace in the database.

File: deploy/production/ingress.yml (hunk 2 above)
Comment (EarthlingDavey) [reply deploy/production/ingress.yml:10]: Would be good to have a chat about this.

I don't think we factored in that we need to have Antony switch over the DNS, **then** get an ssl cert.

We could get around it...

- Create an a record to squiz. 
- Antony switches the DNS to Cloud Platform. 
- Squiz content is still being served.
- Add Certificate.yaml to cloud-platform-environments

File: deploy/production/deployment.tpl.yml (hunk 1 above)
Comment (EarthlingDavey) [reply deploy/production/deployment.tpl.yml:51]: Good point 🤣

**Review by wilson1000 — APPROVED** [review]

**Pull request merged**